rust/lance-core/src/io/writer/statistics.rs

```diff
@@ -495,26 +1345,26 @@ mod tests {
         assert_eq!(
             stats,
             StatisticsRow {
-                null_count: 0_i64,
+                null_count: ScalarValue::from(0_i64),
                 min_value: ScalarValue::from(-10.0_f32),
                 max_value: ScalarValue::from(5.0_f32),
             }
         );
 
         // (Negative) Infinity can be min or max.
-        let arrays: Vec<ArrayRef> = vec![Arc::new(Float32Array::from(vec![
-            4.0f32,
-            std::f32::INFINITY,
-            std::f32::NEG_INFINITY,
+        let arrays: Vec<ArrayRef> = vec![Arc::new(Float64Array::from(vec![
+            4.0f64,
+            std::f64::INFINITY,
+            std::f64::NEG_INFINITY,
         ]))];
         let array_refs = arrays.iter().collect::<Vec<_>>();
         let stats = collect_statistics(&array_refs);
         assert_eq!(
             stats,
             StatisticsRow {
-                null_count: 0_i64,
-                min_value: ScalarValue::from(std::f32::NEG_INFINITY),
-                max_value: ScalarValue::from(std::f32::INFINITY),
+                null_count: ScalarValue::from(0_i64),
+                min_value: ScalarValue::from(std::f64::NEG_INFINITY),
+                max_value: ScalarValue::from(std::f64::INFINITY),
```

Comment:
Can we also test that if the entire array is NaN, then the stats will be `min_value: -Inf, max_value: Inf`?

Reply:
Nice catch. Fixed, but not very elegantly: https://github.com/lancedb/lance/pull/1335/commits/3b6cbc26cb7d48e07cf3431621d0f5402e674460